myst_nb/parser.py

```diff
@@ -39,69 +131,43 @@ def parse(self, inputstring, document):
         output_dir = build_dir.joinpath("jupyter_execute", doc_relpath)
         write_notebook_output(ntbk, str(output_dir), doc_filename)
 
-        # Parse notebook-level metadata as front-matter
-        # For now, only keep key/val pairs that point to int/float/string
-        metadata = ntbk.metadata
-        docinfo = dict_to_docinfo(metadata)
-        document += docinfo
-
         # If there are widgets, this will embed the state of all widgets in a script
         if contains_widgets(ntbk):
             document.append(JupyterWidgetStateNode(state=get_widgets(ntbk)))
-        renderer = SphinxRenderer(document=document, current_node=None)
-        with renderer:
-            # Loop through cells and render them
-            for ii, cell in enumerate(ntbk.cells):
-                # Skip empty cells
-                if len(cell["source"]) == 0:
-                    continue
-                try:
-                    _render_cell(cell, renderer)
-                except Exception as exc:
-                    source = cell["source"][:50]
-                    if len(cell["source"]) > 50:
-                        source = source + "..."
-                    msg_node = self.reporter.error(
-                        (
-                            f"\nError parsing notebook cell #{ii+1}: {exc}\n"
-                            f"Type: {cell['cell_type']}\n"
-                            f"Source:\n{source}\n\n"
-                        )
-                    )
-                    msg_node += nodes.literal_block(cell["source"], cell["source"])
-                    renderer.current_node += [msg_node]
-                    continue
-
-
-def _render_cell(cell, renderer):
-    """Render a cell with a SphinxRenderer instance.
-
-    Returns nothing because the renderer updates itself.
-    """
-    tags = cell.metadata.get("tags", [])
-    if "remove_cell" in tags:
-        return
-
-    # If a markdown cell, simply call the Myst parser and append children
-    if cell["cell_type"] == "markdown":
-        document = Document.read(cell["source"], front_matter=False)
-        # Check for tag-specific behavior because markdown isn't wrapped in a cell
-        if "hide_input" in tags:
-            container = nodes.container()
-            container["classes"].extend(["toggle"])
-            with renderer.current_node_context(container, append=True):
-                renderer.render(document)
-        else:
-            renderer.render(document)
-
-    # If a code cell, convert the code + outputs
-    elif cell["cell_type"] == "code":
+
+        # render the Markdown AST to docutils AST
+        renderer = SphinxNBRenderer(
+            parse_context=parse_context, document=document, current_node=None
```

Comment:
I can't figure out where, but before now the span-level blocks need to have been parsed right?

Reply:
The span level blocks are parsed above in `result.node.children = result.node.children.expand()` (see https://mistletoe-ebp.readthedocs.io/en/latest/using/intro.html#the-parse-process)